app/indexers/collection_indexer.rb

```diff
@@ -31,8 +35,9 @@ def ephemera_collection_titles
 
   def collections
     return [] unless resource.respond_to?(:member_of_collection_ids) && resource.member_of_collection_ids
-    @collections ||=
-      query_service.find_references_by(resource: resource, property: :member_of_collection_ids).to_a.map(&:decorate)
+    @collections ||= resource.member_of_collection_ids.map do |id|
+      query_service.find_by(id: id)&.decorate
```

Comment:
Won't this mean if I have a bad collection ID now it'll error instead of just not index the name?

Reply:
Fixed.